src/file/file.service.ts

```diff
@@ -37,20 +38,27 @@ export class FileService {
    * @param name - of the file uploaded
    * @param ownerID - the id of the file owner
    * @param parent - the folder id in which the file resides
+   * @param size - the size of the file that is being uploaded.
    */
   public static async createUpload(
     key: string,
     bucket: string,
     name: string,
     ownerID: string,
-    parent: string
+    parent: string,
+    size: number = 0,
   ) : Promise<IUpload> {
     const file = await FilesRepository.getFileInFolderByName(parent, name, ownerID);
     if (file) {
       throw new FileExistsWithSameName();
     }
-    const upload: Partial<IUpload> = { key, bucket, name, ownerID, parent };
-    return await UploadRepository.create(upload);
+
+    const createdUpload = await UploadRepository.create({ key, bucket, name, ownerID, parent, size });
+    if (createdUpload) {
+      await QuotaService.updateUsed(ownerID, size);
+    }
```

Comment:
What happens if the file upload fails (after the `createUpload`)?
There should be a function that updates it back